source/meta.tex

```diff
@@ -2554,6 +2570,3461 @@
+If \tcode{b} represents a direct base class relationship of an empty base class,
+then \tcode{size_of(b) > 0}.
+%FIXME: "is true"
+\end{note}
+\end{itemdescr}
+
+\indexlibraryglobal{alignment_of}%
+\begin{itemdecl}
+consteval size_t alignment_of(info r);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constantwhen
+\tcode{dealias(r)} is a reflection of a
+type,
+object,
+value,
+variable of non-reference type,
+non-static data member that is not a bit-field,
+direct base class relationship, or
+data member description.
+If \tcode{dealias(r)} represents a type,
+then \tcode{is_complete_type(r)} is \tcode{true}.
+
+\pnum
+\returns
+\begin{itemize}
+\item
+  If \tcode{dealias(r)} represents a type $T$,
+  then the alignment requirement for the layout-associated type\iref{class.mem.general}
+  for a non-static data member of type $T$.
+\item
+  Otherwise, if \tcode{dealias(r)} represents a variable or object,
+  then the alignment requirement of the variable or object.
+\item
+  Otherwise, if \tcode{r} represents a direct base class relationship,
+  then \tcode{alignment_of(type_of(r))}.
+\item
+  Otherwise, if \tcode{r} represents a non-static data member,
+  then the alignment requirement of the subobject
+  associated with the represented entity
+  within any object of type \tcode{parent_of(r)}.
+\item
+  Otherwise, \tcode{r} represents a data member description
+  $(\mathit{TR}, N, A, W, \mathit{NUA})$\iref{class.mem.general}.
+  If $A$ is not $\perp$,
+  then the value of $A$.
+  Otherwise, \tcode{alignment_of(\reflexpr{$T$})}.
+\end{itemize}
+\end{itemdescr}
+
+\indexlibraryglobal{bit_size_of}%
+\begin{itemdecl}
+consteval size_t bit_size_of(info r);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constantwhen
+\tcode{dealias(r)} is a reflection of a
+type,
+object,
+value,
+variable of non-reference type,
+non-static data member,
+unnamed bit-field,
+direct base class relationship, or
+data member description.
+If \tcode{dealias(r)} represents a type $T$,
+there is a point within the evaluation context from which $T$ is not incomplete.
+
+\pnum
+\returns
+\begin{itemize}
+\item
+  If \tcode{r} represents a non-static data member
+  that is a bit-field
+  %FIXME: comma. I believe non-static data members cannot be unnamed bit-fields,
+  %so the comma would prevent the sentence from suggesting that false reality.
+  or an unnamed bit-field with width $W$,
+  then $W$.
+\item
+  Otherwise, if \tcode{r} represents a data member description
+  $(T, N, A, W, \mathit{NUA})$\iref{class.mem.general}
+  and $W$ is not $\perp$,
+  then $W$.
+\item
+  Otherwise, \tcode{CHAR_BIT * size_of(r)}.
+\end{itemize}
+\end{itemdescr}
+
+\rSec2[meta.reflection.extract]{Value extraction}
+
+\pnum
+The \tcode{extract} function template may be used
+to extract a value out of a reflection when its type is known.
+
+\pnum
+The following are defined for exposition only
+to aid in the specification of \tcode{extract}:
+%FIXME: does ISO like paragraphs ending in a colon?
```

Comment:
so-so.

I don't think a colon is needed here, so please replace with a period.